performance/build.gradle

```diff
@@ -137,8 +187,51 @@ class Results {
     }
 }
 
+class JsonSerializable {
+    // Standard groovy classes have extra fields contentHash and originalClassName
+    protected def instance = new Object()
+    
+    String toJson() {
+        return new JsonBuilder(instance).toPrettyString()
+    }
+    
+    Object getInstance() {
+        return instance
+    }
+}
+
+class Machine extends JsonSerializable {
+    Machine() {
+        instance.metaClass.os =  System.getProperty("os.name") // OperatingSystem.current().getName()
```

Comment:
Decided to do in one way as other found ways don't cover all needed information.  Saved possible wrappers in comments.